plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3326,4 +3327,51 @@ public void testOptimizeForPartitionedTable()
         assertThat(updatedFiles).hasSize(3);
         assertThat(getAllDataFilesFromTableDirectory(tableName)).containsExactlyInAnyOrderElementsOf(concat(initialFiles, updatedFiles));
     }
+
+    @Test
+    public void testVacuumParameterValidation()
+    {
+        assertQueryFails(
+                "ALTER TABLE no_such_table_exists EXECUTE VACUUM",
+                "\\Qline 1:1: Table 'iceberg.tpch.no_such_table_exists' does not exist");
+        assertQueryFails(
+                "ALTER TABLE nation EXECUTE VACUUM (retention_threshold => '33')",
+                "\\QUnable to set procedure property 'retention_threshold' to ['33']: duration is not a valid data duration string: 33");
+        assertQueryFails(
+                "ALTER TABLE nation EXECUTE VACUUM (retention_threshold => '33mb')",
+                "\\QUnable to set procedure property 'retention_threshold' to ['33mb']: Unknown time unit: mb");
+    }
+
+    @Test
```

Comment:
can you add a check at the file level to see whether the vacuum operation was effective?

You could use `$files` metadata table (while the snapshot to be expired is still active)  to gather the file(s) relevant for the snapshot.
You could use `$snapshots` / `$history` metadata tables to actually check that the snapshot was expired.
Subsequently you could check at the file level that the files you previously recorded are also removed.

Reply:
I added counting physical files as it is more reliable